test/e2e/all_in_one_test.go

```diff
@@ -336,52 +339,60 @@ func TestDeployAllInOneDBLESS(t *testing.T) {
 	t.Log("running ingress tests to verify all-in-one deployed ingress controller and proxy are functional")
 	deployIngress(ctx, t, env)
 	verifyIngress(ctx, t, env)
+	ensureAllProxyReplicasAreConfigured(ctx, t, env, deployments.ProxyNN)
 
-	const replicasCount = 3
-	gatewayDeployment := deployments.GetProxy(ctx, t, env)
-	gatewayDeployment.Spec.Replicas = lo.ToPtr(int32(replicasCount))
-	_, err := env.Cluster().Client().AppsV1().Deployments(gatewayDeployment.Namespace).Update(ctx, gatewayDeployment, metav1.UpdateOptions{})
-	require.NoError(t, err)
+	expectedControllerReplicas := *(deployments.GetController(ctx, t, env).Spec.Replicas)
+	t.Logf("store expected controller replicas count = %d", expectedControllerReplicas)
```

Comment:
Is it necessary to store the expected controller replicas? Did you intend to scale the controller deployment here?

Reply:
Good catch, yeah, it's not used. I meant to use it to ensure that after scaling the proxy down to 0, the count of the controller's ready pods doesn't change. Fixed that: d66c33d